### PR TITLE
advancedtls: Rename VType

### PR DESCRIPTION
**Pull request by gtcooke94**
This PR renames `VType` to `VerificationType` for clarity.
It marks `VType` as deprecated. For now, if `VType` is explicitly set (does not have the default value) then we set `VerificationType` to `VType` so that user behavior doesn't break.

RELEASE NOTES: none

### CODE REVIEW

**Review by erm-g — APPROVED** [review]
LGTM

File: security/advancedtls/advancedtls.go [hunk 1]
```diff
@@ -175,6 +175,11 @@ type ClientOptions struct {
 	// RootOptions is OPTIONAL on client side. If not set, we will try to use the
 	// default trust certificates in users' OS system.
 	RootOptions RootCertificateOptions
+	// VerificationType defines what type of server verification is done. See
+	// the `VerificationType` enum for the different options.
+	// Default: CertAndHostVerification
+	VerificationType VerificationType
+	// DEPRECATED: Renamed to `VerificationType`
 	// VType is the verification type on the client side.
```
Comment (dfawley) [inline security/advancedtls/advancedtls.go:183]: The proper way to docstring this is:

```go
// VType is blah blah
//
// Deprecated: use VerificationType instead.
VType VerificationType
```

File: security/advancedtls/advancedtls.go [hunk 2]
```diff
@@ -210,6 +215,11 @@ type ServerOptions struct {
 	RootOptions RootCertificateOptions
 	// If the server want the client to send certificates.
 	RequireClientCert bool
+	// VerificationType defines what type of client verification is done. See
+	// the `VerificationType` enum for the different options.
+	// Default: CertAndHostVerification
+	VerificationType VerificationType
+	// DEPRECATED: Renamed to `VerificationType`
```
Comment (dfawley) [inline security/advancedtls/advancedtls.go:222]: As above.

File: security/advancedtls/advancedtls.go (hunk 1 above)
Comment (gtcooke94) [reply security/advancedtls/advancedtls.go:183]: Fixed

File: security/advancedtls/advancedtls.go (hunk 2 above)
Comment (gtcooke94) [reply security/advancedtls/advancedtls.go:222]: Fixed

**Review by dfawley — APPROVED** [review]

**Pull request merged**